server/src/main/java/org/elasticsearch/search/SearchPhase.java

```diff
@@ -35,4 +36,32 @@
      * Executes the search phase
      */
     void execute(SearchContext context);
+
+    class SearchContextRequestLog {
```

Comment:
I think this can be declared as `static` to avoid capturing the enclosing class' state? Further looking at the usage, I wonder whether this should actually be a static method to avoid creating garbage for every query?

Reply:
This declaration is within an interface so the `static` modifier is redundant, this is a static nested class already.

Reply:
Ah, good point. I did not check that.